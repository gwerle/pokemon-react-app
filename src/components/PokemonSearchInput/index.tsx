import React, { useState } from "react";

import TextInput from "../base/TextInput";

type Props = {
  setSearchValue: (value: string) => void;
};

export default function PokemonSearchInput({ setSearchValue }: Props) {
  const [value, setValue] = useState("");

  const handleSubmit = (e: { preventDefault: () => void }) => {
    e.preventDefault();

    setSearchValue(value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <TextInput
        label="Pesquise o nome do seu PokÃ©mon favorito:"
        value={value}
        buttonType="submit"
        onInput={setValue}
      />
    </form>
  );
}
