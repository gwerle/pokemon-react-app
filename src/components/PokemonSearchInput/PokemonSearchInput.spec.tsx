import { screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { renderWithTheme } from "@/utils/tests/renderWithTheme";

import PokemonSearchInput from ".";

describe("<PokemonList />", () => {
  it("Renders without error", () => {
    const setSearchValueFn = jest.fn();
    const { container } = renderWithTheme(
      <PokemonSearchInput
        setSearchValue={setSearchValueFn}
      ></PokemonSearchInput>
    );

    expect(container).toMatchSnapshot();
  });

  it("should call function when submit", async () => {
    const setSearchValueFn = jest.fn();
    renderWithTheme(
      <PokemonSearchInput
        setSearchValue={setSearchValueFn}
      ></PokemonSearchInput>
    );

    const input = screen.getByTestId("pokemon-search");
    const text = "This is a text{enter}";
    userEvent.type(input, text);

    await waitFor(() => {
      expect(setSearchValueFn).toHaveBeenCalledTimes(1);
    });
  });
});
