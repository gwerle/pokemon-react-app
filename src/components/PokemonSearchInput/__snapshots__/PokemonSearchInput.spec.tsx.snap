// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonList /> Renders without error 1`] = `
<div>
  <form>
    <div
      class="sc-jTrPJq hawfGP"
    >
      <label
        class="sc-dmqHEX nccCQ"
      >
        Pesquise o nome do seu Pok√©mon favorito:
      </label>
      <div
        class="sc-beqWaB kqrFeq"
      >
        <input
          class="sc-gueYoa eemTcj"
          data-testid="pokemon-search"
          type="text"
          value=""
        />
        <button
          class="sc-eDDNvR eldutF"
          type="submit"
        >
          <svg
            fill="currentColor"
            height="16px"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
            />
            <path
              d="M11.412 8.586c.379.38.588.882.588 1.414h2a3.977 3.977 0 0 0-1.174-2.828c-1.514-1.512-4.139-1.512-5.652 0l1.412 1.416c.76-.758 2.07-.756 2.826-.002z"
            />
          </svg>
        </button>
      </div>
    </div>
  </form>
</div>
`;
